tests/test_conflicting_formatters.py

```diff
@@ -0,0 +1,98 @@
+import tempfile
+import tokenize
+from typing import Generator
+
+import pytest
+from _pytest.fixtures import SubRequest
+
+from pydocstringformatter._formatting.base import Formatter
+
+
+class MakeAFormatter(Formatter):
+    """A formatter that makes Bs into As."""
+
+    name = "make-a-formatter"
+    style = ["default"]
+
+    def treat_token(self, tokeninfo: tokenize.TokenInfo) -> tokenize.TokenInfo:
+        """Replace Bs with As."""
+        token_dict = tokeninfo._asdict()
+        token_dict["string"] = token_dict["string"].replace("B", "A")
+        return type(tokeninfo)(**token_dict)
+
+
+class MakeBFormatter(Formatter):
+    """A formatter that makes As into Bs."""
+
+    name = "make-b-formatter"
+    style = ["default"]
+
+    def treat_token(self, tokeninfo: tokenize.TokenInfo) -> tokenize.TokenInfo:
+        """Replace As with Bs."""
+        token_dict = tokeninfo._asdict()
+        token_dict["string"] = token_dict["string"].replace("A", "B")
+        return type(tokeninfo)(**token_dict)
+
+
+class AddBFormatter(Formatter):
+    """A formatter that adds Bs."""
+
+    name = "add-b-formatter"
+    style = ["default"]
+
+    def treat_token(self, tokeninfo: tokenize.TokenInfo) -> tokenize.TokenInfo:
+        """Add a B to the end of the string."""
+        token_dict = tokeninfo._asdict()
+        token_dict["string"] += "B"
+        return type(tokeninfo)(**token_dict)
+
+
+@pytest.fixture(scope="function")
+def patched_run(request: SubRequest) -> Generator[type, None, None]:
```

Comment:
```suggestion
def patched_run(request: SubRequest) -> Generator[type[_Run], None, None]:
```
right?

Reply:
Indeed! To be completely honest I did not know how to annotate "the class itself and not an instance of it". Thanks